### PR TITLE
[containerapp] Give a better error message when customer uses --artif…

### PR DESCRIPTION
**Pull request by daniv-msft**
…act with ACR Task

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/containerapp/azext_containerapp/_up_utils.py [hunk 1]
```diff
@@ -582,6 +582,10 @@ def build_container_from_source_with_acr_task(self, image_name, source):
         from azure.cli.command_modules.acr._client_factory import cf_acr_tasks, cf_acr_runs
         from azure.cli.core.profiles import ResourceType
 
+        # Validate that the source provided is a directory, and not a file.
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_up_utils.py:585]: In the preview code, the `source` value is infer from `artifact`, the customer not input `--source`, but can see this error when `docker` is not running .

**Comment by yonzhan** [comment]
containerapp

File: src/containerapp/azext_containerapp/_up_utils.py [hunk 2]
```diff
@@ -582,6 +582,10 @@ def build_container_from_source_with_acr_task(self, image_name, source):
...
+        if os.path.isfile(source):
+            raise ValidationError(f"Impossible to build the artifact file {source} with ACR Task. Please make sure that you use --source and target a directory, or if you want to build your artifact locally that Docker is running on your machine.")
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_up_utils.py:587]:
```suggestion
            raise ValidationError(f"Impossible to build the artifact file {source} with ACR Task. Please make sure that you use --source and target a directory, or if you want to build your artifact locally, please make sure Docker is running on your computer.")
```

Comment (daniv-msft) [reply src/containerapp/azext_containerapp/_up_utils.py]: Thanks, I just updated it

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
Could you help to merge this PR before help to merge the PR for review version 0.3.44 https://github.com/Azure/azure-cli-extensions/pull/6972 ?Thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**